app/interactors/donation/create.rb

```diff
@@ -0,0 +1,17 @@
+class Donation
+  include Charge
+  include Customer
+  include Token
+
+  class Create
+    include Interactor::Organizer
+
+    organize Token::Generate, Customer::Create, Charge::Create
```

Comment:
Yes, absolutely
The `Donation::Create` interactor works directly with Stripe so there's nothing persisted to the database at this stage hence no need to rollback, everything else is up to you :)
Quite flexible in my opinion